luxjs/forms/utils.js

```diff
@@ -3,54 +3,290 @@
  */
 
 angular.module('lux.form.utils', ['lux.services'])
+    //
+    .factory('remoteService', ['$lux', '$q', '$timeout', function($lux, $q, $timeout) {
+        return {
+            /**
+             * Called to get remote options from the API
+             *
+             * @param api
+             * @param target
+             * @param scope
+             * @param attrs
+             * @param config {object} - parameters passed to query string in request
+             * @param searchValue {string} - value of the search was were typed in input field
+             * @param extendCurrentOptions {boolean} - flag that indicates whether to add new options to existing options
+             * @returns {promise}
+             */
+            query: function(api, target, scope, attrs, config, searchValue, extendCurrentOptions) {
+                var defer = $q.defer();
+
+                if (extendCurrentOptions) {
+                    // Extend current options
+                    options = scope[target.name];
+
+                    if (options.length > config.optionsTotal) {
+                        defer.reject();
+                        return defer.promise;
+                    } else {
+                        // Increase offset
+                        config.params.offset += config.params.limit;
+                    }
+                } else {
+                    // Add initial options
+                    var options = [];
+                    scope[target.name] = options;
+
+                    config.initialValue.id = '';
+                    config.initialValue.name = 'Loading...';
+
+                    options.push(config.initialValue);
+                }
+
+                if (searchValue === undefined)
+                    delete config.params[config.id];
+                else
+                    config.params[config.id] = searchValue;
 
-    .directive('remoteOptions', ['$lux', function ($lux) {
+                api.get(null, config.params).then(function(data) {
+                    // Get amount of total items
+                    config.optionsTotal = data.data.total;
+
+                    if (searchValue !== undefined && data.data.result.length === 0) {
+                        options[0].name = 'Cannot find value';
```

Comment:
If there are no search results, the message should something like `No matches found`.